### ISSUE TITLE
rxjs/finnish doesn't detect implicitly typed Observable functions

### ISSUE THREAD

**Issue by leroydev** [cartant/eslint-plugin-rxjs#141]
The `rxjs/finnish` rule flags me to end `FunctionExplicitlyReturningObservable` with a `$` but doesn't do the same for `FunctionImplicitlyReturningObservable`. It seems like this is because the rule doesn't detect functions implicitly having an Observable return type?
```ts
import { Observable, of } from 'rxjs';

function FunctionImplicitlyReturningObservable() {
  return of([]);
}

function FunctionExplicitlyReturningObservable(): Observable<[]> {
  return of([]);
}
```
<img width="809" height="357" alt="Image" src="https://github.com/user-attachments/assets/ee310fd7-8cb0-4f44-9005-b24fefbde3c9" />

If this is seen as a problem I could try to make a PR to fix this.

Thanks in advance!

**Comment by JasonWeinzierl** [comment]
Hello,

My fork `eslint-plugin-rxjs-x` has a fix published for this as of https://github.com/JasonWeinzierl/eslint-plugin-rxjs-x/pull/185 .
